scheduler/cups-driverd.cxx

```diff
@@ -2914,3 +2927,74 @@ regex_string(const char *s)		/* I - String to compare */
 
   return (NULL);
 }
+
+void *
+cupsFileGetsCall(void *input)
+{
+  retval = cupsFileGets(((struct args*)input)->fp,((struct args*)input)->buf,((struct args*)input)->buflen);
+  return NULL;
+}
+
+/*
+  * Alternate to cupsFileGets() 
+  */
+
+char * 
+cupsFileGetsT(cups_file_t *fp,		/* I - CUPS file */
+             char        *buf,		/* O - String buffer */
+	     size_t      buflen)        /* I - Size of string buffer */
+{
+  
+  int flag = 0;
+
+ /*
+  * Creating a new thread ID.
+  */
+
+  pthread_t ptid;
+  struct timeval t1, t2;
+
+  struct args *arg = (struct args *)malloc(sizeof(struct args));
+  arg->fp = fp;
+  arg->buf = buf;
+  arg->buflen = buflen;
+
+  gettimeofday(&t1,NULL);
+
+  pthread_create(&ptid,NULL,cupsFileGetsCall,(void*)arg);
+
+ /*
+  * Loop through cupsFileGets() until timeout (2 sec)
+  */
+
+  while(retval == NULL)
+  {
+    gettimeofday(&t2, NULL);
+    double elapsedTime = (t2.tv_sec - t1.tv_sec);
+
+    if(elapsedTime > 2)
+    {
+      flag=1;
+      break;
+    }
+
+    sleep(0.5);
+  }
+
+  if(flag)
+  {
+
+   /*
+    * Terminating cupsFileGets() thread due to timeout.
+    */
+
+    fprintf(stderr,"%s\n","Thread terminated");
+    pthread_cancel(ptid);
+
+    retval = NULL;
+  }
+  else
+    pthread_join(ptid, NULL);
+
+return retval;
+}
```

Comment:
@pranjanpr missing newline on the end of the file - it's common to end a file with new line.